components/org.wso2.bfsi.consent.management.service/src/main/java/org/wso2/bfsi/consent/management/service/impl/ConsentCoreServiceImpl.java

```diff
@@ -0,0 +1,1905 @@
+                        + consentID);
+            }
+        } finally {
+            log.debug(ConsentCoreServiceConstants.DATABASE_CONNECTION_CLOSE_LOG_MSG);
+            DatabaseUtil.closeConnection(connection);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean revokeExistingApplicableConsents(String clientID, String userID, String consentType,
+                                                    String applicableStatusToRevoke,
+                                                    String revokedConsentStatus, boolean shouldRevokeTokens)
+            throws ConsentManagementException {
+
+        if (StringUtils.isBlank(clientID) || StringUtils.isBlank(revokedConsentStatus) || StringUtils.isBlank(userID)
+                || StringUtils.isBlank(applicableStatusToRevoke) || StringUtils.isBlank(consentType)) {
+            log.error(ConsentCoreServiceConstants.REVOKE_DETAILS_MISSING_ERROR_MSG);
+            throw new ConsentManagementException(ConsentCoreServiceConstants.REVOKE_DETAILS_MISSING_ERROR_MSG);
+        }
+
+        Connection connection = DatabaseUtil.getDBConnection();
+
+        try {
+            ConsentCoreDAO consentCoreDAO = ConsentStoreInitializer.getInitializedConsentCoreDAOImpl();
+            try {
+
+                ArrayList<String> accountMappingIDsList = new ArrayList<>();
+                ArrayList<String> clientIDsList = ConsentCoreServiceUtil.constructArrayList(clientID);
+                ArrayList<String> userIDsList = ConsentCoreServiceUtil.constructArrayList(userID);
+                ArrayList<String> consentTypesList = ConsentCoreServiceUtil.constructArrayList(consentType);
+                ArrayList<String> consentStatusesList = ConsentCoreServiceUtil
+                        .constructArrayList(applicableStatusToRevoke);
+
+                // Get existing consents
+                log.debug("Retrieving existing consents");
+
+                // Only parameters needed for the search are provided, others are made null
+                ArrayList<DetailedConsentResource> retrievedDetailedConsentResources = consentCoreDAO
+                        .searchConsents(connection, null, clientIDsList, consentTypesList,
+                                consentStatusesList, userIDsList, null, null, null, null);
+
+                // Revoke existing consents and create audit records
+                for (DetailedConsentResource resource : retrievedDetailedConsentResources) {
+                    String previousConsentStatus = resource.getCurrentStatus();
+
+                    // Update consent status
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("Updating consent status for consent ID: %s",
+                                resource.getConsentID().replaceAll("[\r\n]", "")));
+                    }
+                    consentCoreDAO.updateConsentStatus(connection, resource.getConsentID(), revokedConsentStatus);
+
+                    //TODO
```

Comment:
we can remove this right?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/3/commits/08d281c66fc90f979dcae36a8c8baedf99ab9aa9